clang/lib/Frontend/ASTUnit.cpp

```diff
@@ -1243,12 +1247,19 @@ bool ASTUnit::Parse(std::shared_ptr<PCHContainerOperations> PCHContainerOps,
   }
 
   std::unique_ptr<TopLevelDeclTrackerAction> Act(
-      new TopLevelDeclTrackerAction(*this));
+      new TopLevelDeclTrackerAction(*this, true));
```

Comment:
We should use [bugprone-argument-comment](https://clang.llvm.org/extra/clang-tidy/checks/bugprone/argument-comment.html) comments on literals passed as arguments.

```suggestion
      new TopLevelDeclTrackerAction(*this, /*_ReusePreprocessor=*/true));
```

Reply:
Moreover, can you explain the "reuse  preprocessor" logic, which seems orthogonal to the goal of preserving `__COUNT__`? (ie, i would expect count to be preserved, not the whole preprocessor).
Unless I am missing something, I think this would simplify the patch a bit.

Reply:
The problem is that when you are using PCH it always recreate the preprocessor for compiling the .c/.cpp file instead of reusing the one that resulted from the PCH.  So we need a way to ensure that it isn't recreated in this process.